src/workers/postgres.rs

```diff
@@ -299,23 +343,40 @@ impl Postgres {
         tokio::spawn(async move {
             info!("Writing to postgres");
 
-            let mut tx_que = Vec::<PostgresTx>::new();
-            let mut block_que = Vec::new();
-            let mut update_que = Vec::new();
+            const TX_MAX_CAPACITY: usize = get_max_safe_inserts::<PostgresTx>();
+            const BLOCK_MAX_CAPACITY: usize = get_max_safe_inserts::<PostgresBlock>();
+            const UPDATE_MAX_CAPACITY: usize = get_max_safe_updates::<PostgresUpdateTx>();
+
+            let mut tx_batch: Vec<PostgresTx> = Vec::with_capacity(TX_MAX_CAPACITY);
+            let mut block_batch: Vec<PostgresBlock> = Vec::with_capacity(BLOCK_MAX_CAPACITY);
+            let mut update_batch =
+                Vec::<(PostgresUpdateTx, String)>::with_capacity(UPDATE_MAX_CAPACITY);
+
+            let mut session_establish_error = false;
 
             loop {
                 loop {
```

Comment:
Could move all the vectors in the loop so that they are cleared at the end of the loop.

Reply:
no we don't want reallocation of memory. After a fixed capacity is allocated, just clear it and re use it. More efficient.